ibis/expr/types/relations.py

```diff
@@ -247,6 +210,40 @@ def _bind_reduction_filter(self, where):
 
         return where.resolve(self)
 
+    def bind(self, *args, **kwargs):
+        # allow the first argument to be either a dictionary or a list of values
+        if len(args) == 1:
```

Comment:
I'm a -.5 on this special casing. This causes `t.bind([1,"a"])`  to be `(ibis.literal(1), t.a)` (behavior A). I would prefer for it to be `ibis.array([1, "a"])` (behavior B). I looked back through this PR to try to get the context, but I might not have gotten all of it. As I understand it, switching A to B

- Would be the preferred behavior by everyone else if we were starting from a clean slate with no history.
- Might silently change the behavior of existing user code (we could add an easily-silencable warning for the 9.0 and maybe 10.0 versions?). This is the main downside.
- Would this cause any other paper cuts or make impossible some other task I'm not thinking of?

Reply:
> Might silently change the behavior of existing user code (we could add an easily-silencable warning for the 9.0 and maybe 10.0 versions?).

The benefit of behavior B doesn't seem worth the pain this would cause given history.

I don't think changing to behavior B makes anything impossible, but it's a breaking API change that seems unnecessary.

Reply:
> Would be the preferred behavior by everyone else if we were starting from a clean slate with no history.

It would be my preference, but yes, that would be too big of a breakage. If we wish to change it we can also do that for later releases.

Reply:
sounds good let's keep it as is for now. Thank you!